bot/src/discord_commands/schedule.js

```diff
@@ -0,0 +1,165 @@
+const { Permissions } = require('monochrome-bot');
+const randomHelper = require('./random.js');
+
+let intervals = [];
+
+const NUMBER_OF_RETRIES = 50;
+
+function randomAction(msg, suffix, monochrome) {
+  const suffixLowerCase = suffix.toLowerCase();
+  monochrome.updateUserFromREST(msg.author.id).catch(() => {});
+
+  return randomHelper.getRandomWordRecursive(
+    suffixLowerCase,
+    msg,
+    NUMBER_OF_RETRIES,
+    monochrome,
+  );
+}
+
+function setTimer(id, msg, level, monochrome, freq, startTime) {
+  let nextTime = 0;
+  if (startTime) {
+    nextTime = startTime.getTime() - (new Date()).getTime();
+    while (nextTime < 0) {
+      nextTime += freq;
+    }
+  } else {
+    nextTime = freq;
+  }
+  let index = intervals.findIndex(i => i.id === id);
+  if (index !== -1) {
+    intervals[index].interval = setTimeout(() => {
+      randomAction(msg, level, monochrome);
+      setTimer(id, msg, level, monochrome, freq);
+    }, nextTime)
+  }
+}
+
+async function setSchedule(suffix, msg, monochrome) {
+  let freq = '';
+  let freqType = '';
+  let start = '';
+  let level = '';
+  let suffixArray = suffix.split(" ");
+  if (suffixArray.length === 1 && suffixArray[0] === 'clear') {
+    intervals.forEach(i => clearTimeout(i.interval));
+    intervals = [];
+    msg.channel.createMessage('All scheduled commands have been cleared.');
+  } else if (suffixArray.length === 1 && suffixArray[0] === 'list') {
+    if (intervals.length > 0) {
+      let list = '';
+      intervals.forEach(i => list = (list === '' ? '' : list + '\n') + 'ID: ' + i.id +
+                                            ', Channel: ' + i.channel +
+                                            ', Frequency: ' + i.frequency +
+                                            ', Start time: ' + i.start +
+                                            ', Level: ' + i.level);
+      msg.channel.createMessage(list);
+    } else {
+      msg.channel.createMessage('List is empty');
+    }
+  } else if (suffixArray.length === 2 && suffixArray[0] === 'stop') {
+    let index = intervals.findIndex(i => i.id = suffixArray[1]);
```

Comment:
Is this supposed to be `i.id === suffixArray[1]`? Otherwise this will always return 0 (or -1 if the array is empty).

Reply:
Absolutely. My mistake.